### ISSUE TITLE
Unable to receive sourcePackage with large sourceCode attribute

### ISSUE THREAD

**Issue by vitormsilva** [reTHINK-project/dev-catalogue#25]
Hi,

I have a problem, and need help to solve it;

When i load into the catalogue-database the [RuntimeUA distribution file](https://github.com/reTHINK-project/dev-runtime-core/blob/dev-0.4/dist/RuntimeUA.js), the catalogue-broker brokes, and throw an error:

In the log console i have this:
```shell
 sudo docker logs --tail=18 catalogue-broker
	at org.eclipse.leshan.tlv.TlvDecoder.decode(TlvDecoder.java:143) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	at org.eclipse.leshan.tlv.TlvDecoder.decode(TlvDecoder.java:127) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	at org.eclipse.leshan.core.node.codec.LwM2mNodeDecoder.decode(LwM2mNodeDecoder.java:98) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	at org.eclipse.leshan.server.californium.impl.LwM2mResponseBuilder.buildContentResponse(LwM2mResponseBuilder.java:265) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	... 28 more
Caused by: org.eclipse.leshan.tlv.TlvException: Invalid 'value' length
	at org.eclipse.leshan.tlv.TlvDecoder.decode(TlvDecoder.java:135) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	at org.eclipse.leshan.tlv.TlvDecoder.decode(TlvDecoder.java:127) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	at org.eclipse.leshan.core.node.codec.LwM2mNodeDecoder.decode(LwM2mNodeDecoder.java:98) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	at org.eclipse.leshan.server.californium.impl.LwM2mResponseBuilder.buildContentResponse(LwM2mResponseBuilder.java:265) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	... 28 more
Caused by: java.lang.IllegalArgumentException
	at java.nio.Buffer.limit(Buffer.java:267) ~[?:1.7.0_95]
	at org.eclipse.leshan.tlv.TlvDecoder.decode(TlvDecoder.java:125) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	at org.eclipse.leshan.tlv.TlvDecoder.decode(TlvDecoder.java:127) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	at org.eclipse.leshan.core.node.codec.LwM2mNodeDecoder.decode(LwM2mNodeDecoder.java:98) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	at org.eclipse.leshan.server.californium.impl.LwM2mResponseBuilder.buildContentResponse(LwM2mResponseBuilder.java:265) ~[rethink-catalogue-broker-1.0.4-jar-with-dependencies.jar:?]
	... 28 more
```
You can see more on image.

![screenshot from 2016-04-18 12-26-38](https://cloud.githubusercontent.com/assets/10596749/14602732/6d3fac30-0561-11e6-96c1-7504317a38a3.png)

The client test: https://catalogue.hybroker.rethink.ptinovacao.pt/client/

Can you try to load the RuntimeUA in your test bed and try, i don't know if the problem is related with the size of file (yes, it is a big file) or it is other problem.

Can you help me?

**Comment by Endebert** [comment]
Hi @vitormsilva,

yes this bug is related to file size and is a bug inside of leshan. 
I contacted them and it will be fixed in their next version, which will hopefully release this month. I'm currently adapting the catalogue to use the new version and the problem will be fixed in version 1.1.0 of the catalogue.

The bug prevents you from retrieving the sourcePackage instance from the broker, but directly requesting the sourceCode should still work, e.g.

http://localhost/.well-known/sourcepackage/XXXXX fails, but
http://localhost/.well-known/sourcepackage/XXXXX/sourceCode works.

**Comment by vitormsilva** [comment]
Hi @Endebert,

Regarding this issue, the problem doesn't happen only with the large files, but with small files too;

You can test https://catalogue.hybroker.rethink.ptinovacao.pt/.well-known/sourcepackage/10002, for example.

When have we an update to this?

Thanks.

**Comment by Endebert** [comment]
I just pushed the new version (1.1.0) into the dev branch, and the relase is currently planned for next tuesday.

If this bug is blocking your progress, please try using the current state of the dev branch, which should work properly.

**Comment by emmelmann-fokus** [comment]
@Endebert , could you just please confirm that you pushed the new release in the "DEV" branch?

I will create additional docker images (with dev tag based on the newest dev branch) right now to help @vitormsilva 

Best

Marc

**Comment by Endebert** [comment]
yes it is the dev branch, I merged the leshan-M10 branch into it.

**Comment by emmelmann-fokus** [comment]
done.  dev tags are available.

I will have to verify that the docker-files in the dev branch do not derive from the master branch.  But this requires concentration and its 1AM right now for me.  I'll look at this over the weekend and we will have tested docker files for the dev branch on Monday morning your time.

New release in the master branch will be on Tuesday your time.

**Comment by Endebert** [comment]
[Fixed in new release 1.1.0.](https://github.com/reTHINK-project/dev-catalogue/releases/tag/1.1.0)
